crates/polars-sql/src/context.rs

```diff
@@ -535,40 +535,44 @@ impl SQLContext {
             if query.order_by.is_empty() {
                 lf.select(projections)
             } else if !contains_wildcard {
+                let mut retained_names = Vec::with_capacity(projections.len());
```

Comment:
We can use `PlIndexSet` if order is important but still want fast lookups and removals.

Reply:
In this case we're just building a Vec to pass to `select`

Reply:
Yup, `exclude` benefits from fast removes - modified accordingly!✌️